java/fury-core/src/main/java/org/apache/fury/serializer/EnumSerializer.java

```diff
@@ -40,20 +50,54 @@ public EnumSerializer(Fury fury, Class<Enum> cls) {
       Preconditions.checkArgument(enclosingClass.isEnum());
       enumConstants = enclosingClass.getEnumConstants();
     }
+
+    if (fury.getConfig().serializeEnumByName()) {
+      // as we know the size of enum is fixed, initialize the size of map with that value
+      int initialCapacity = (int) Math.ceil(enumConstants.length / 0.99f);
+
+      metaStringtoEnumRepresentation = new HashMap<>(initialCapacity, 0.99f);
+
+      for (Enum enumConstant : enumConstants) {
+        if (enumConstant != null) {
+          MetaString ms =
+              Encoders.GENERIC_ENCODER.encode(enumConstant.name(), MetaString.Encoding.UTF_8);
```

Comment:
We'd better use `org.apache.fury.meta.MetaStringEncoder#encode(java.lang.String)` instead, it can get smaller encoded size for enum name than utf8 encoding